File: core/src/main/java/com/krypton/cloud/model/FileType.java
package com.krypton.cloud.model;

public enum FileType {
	
	IMAGE_JPG("JPG"),
	IMAGE_PNG("PNG"),
	IMAGE_RAW("RAW"),
	IMAGE_GIF("GIF"),
	ARCHIVE_ZIP("ZIP"),
	ARCHIVE_TAR("TAR"),
	ARCHIVE_7Z("7z"),
	ARCHIVE_RAR("RAR"),
	MKV("MKV"),
	MP3("MP3"),
	MP4("MP4"),
	TXT("TXT"),
	VIM("VIM"),
	JAR("JAR"),
	EXE("EXE"),
	BAT("BAT"),
	DMG("DMG"),
	APK("APK"),
	IMG_EXEC("IMAGE"),
	SHELL("SH"),
	JAVA("JAVA"),
	KOTLIN("KT"),
	HTML("HTML"),
	CSS("CSS"),
	SCSS("SCSS"),
	JAVASCRIPT("JS"),
	TYPESCRIPT("TS"),
	JSX("JSX"),
	TSX("TSX"),
	GRADLE("GRADLE"),
	JSON("JSON"),
	XML("XML"),
	YML("YML"),
	OTHER("OTHER");

	private final String type;

	public String getType() {
		return type;
	}

	FileType(String type) {
		this.type = type;
	}

	public boolean equalsType(String type) {
		return this.type.equals(type);
	}

	public String toString() {
		return this.type;
	}
}

File: common/src/main/kotlin/common/model/FileType.kt
package common.model

enum class FileType private constructor(val type: String) {

    IMAGE_JPG("JPG"),
    IMAGE_PNG("PNG"),
    IMAGE_RAW("RAW"),
    IMAGE_GIF("GIF"),
    ARCHIVE_ZIP("ZIP"),
    ARCHIVE_TAR("TAR"),
    ARCHIVE_7Z("7z"),
    ARCHIVE_RAR("RAR"),
    MKV("MKV"),
    MP3("MP3"),
    MP4("MP4"),
    TXT("TXT"),
    VIM("VIM"),
    JAR("JAR"),
    EXE("EXE"),
    BAT("BAT"),
    DMG("DMG"),
    APK("APK"),
    IMG_EXEC("IMAGE"),
    SHELL("SH"),
    JAVA("JAVA"),
    KOTLIN("KT"),
    HTML("HTML"),
    CSS("CSS"),
    SCSS("SCSS"),
    JAVASCRIPT("JS"),
    TYPESCRIPT("TS"),
    JSX("JSX"),
    TSX("TSX"),
    GRADLE("GRADLE"),
    JSON("JSON"),
    XML("XML"),
    YML("YML"),
    OTHER("OTHER");

    fun equalsType(type: String): Boolean {
        return this.type == type
    }

    override fun toString(): String {
        return this.type
    }
}
